gap/PolygonalComplexes/navigation.gi

```diff
@@ -919,3 +919,296 @@ InstallMethod( FacesWithEdgeProperties,
         return res;
     end
 );
+
+
+#######################################
+###
+###      Common Methods
+###
+InstallMethod(CommonVerticesOfEdgesNC,
+	"for a polygonal complex and two positive integers",
+	[IsVEFComplex,IsPosInt,IsPosInt],
+	function(complex,edge1,edge2)
+		return Intersection(VerticesOfEdge(complex,edge1),VerticesOfEdge(complex,edge2));
+	end
+);
+
+InstallMethod(CommonVerticesOfEdges,
+	"for a polygonal complex and two positive integers",
+	[IsVEFComplex,IsPosInt,IsPosInt],
+	function(complex,edge1,edge2)
+		__SIMPLICIAL_CheckEdge( complex, edge1, "CommonVerticesOfEdges" );
+		__SIMPLICIAL_CheckEdge( complex, edge2, "CommonVerticesOfEdges" );
+		return CommonVerticesOfEdgesNC(complex,edge1,edge2);
+	end
+);
+
+InstallMethod(CommonVerticesOfEdgesNC,
+	"for a polygonal complex and a list",
+	[IsVEFComplex,IsList],
+	function(complex,edgeList)
+		local vertices,e,comb,vertex;
+		if Length(edgeList)>1 then
+			vertices:=[];
+			comb:=Combinations(edgeList,2);
+			vertices:=CommonVerticesOfEdgesNC(complex,comb[1][1],comb[1][2]);
+			for e in [2..Length(comb)] do
+				vertices:=Intersection(vertices,CommonVerticesOfEdgesNC(complex,comb[e][1],comb[e][2]));
+			od;
```

Comment:
This seems a bit inefficient. Instead of iterating over all pairs of edges, you could just iterate over the edges, i.e.
vertices := VerticesOfEdgeNC(edgeList[1])
for e in [2..Length(edgeList)] do
  vertices := Intersection(vertices, VerticesOfEdgeNC(e);
od